ios/Classes/types/ARTClientOptions+AblyFlutterClientOptions.m

```diff
@@ -0,0 +1,16 @@
+#import "ARTClientOptions+AblyFlutterClientOptions.h"
+#import <objc/runtime.h>
+
+@implementation ARTClientOptions (AblyFlutterClientOptions)
+
+- (void) setHasAuthCallback:(BOOL) value {
+    NSNumber *number = [NSNumber numberWithBool: value];
+    objc_setAssociatedObject(self, @selector(hasAuthCallback), number, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
```

Comment:
Oh, yes, even more reasons not to use a category here! Oooohhhh.... 😬

Reply:
167da5cf297b94e8025bb6f27ebc6abbec1075ef